enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,257 @@
+    metadata:
+      name: cluster-monitoring-config
+      namespace: openshift-monitoring
+    data:
+      config.yaml: |
+        metricsServer:
+          enabled: true
+    ```
+
+With this configuration, the cluster monitoring operator should:
+  1. Deploy the metrics-server resources in the `openshift-monitoring` namespace
+  1. Verify that the `metrics-server` service is functional by querying the `/readyz` endpoint
+  1. Wait for `5 x metrics-server's scrape interval` to have enough data populated in `metrics-server` in-memory
+  1. Update `APIService` resource to point to `metrics-server` API instead of `prometheus-adapter` API
+  1. Delete `prometheus-adapter` resources
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- We propose to add type `metricsServer` to the `openshift-montoring/cluster-monitoring-config` configmap to define 
+  metrics-server settings. 
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    ...
+    // `MetricsServerConfig` defines settings for the MetricsServer component.
+    MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+  }
+  ```
+
+  When `enabled` is set `true`, the cluster monitoring operator configures the Metrics Server resources in `openshift-monitoring` namespace and tears down the Prometheus adapter resources once Metrics Server is fully functional.
+
+  ```go
+  // The `MetricsServerConfig` resource defines settings for the MetricsServer component.
+  type MetricsServerConfig struct {
+    // `Enabled` is a Boolean flag that enables or disables metrics-server.
+    Enabled *bool `json:"enabled,omitempty"`
+    // Defines the nodes on which the pods are scheduled.
+    NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+    // Defines tolerations for the pods.
+    Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+    // Defines resource requests and limits for the Prometheus container.
+    Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+  }
+  ```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Configuration/Enablement
+
+Metrics Server is not enabled by default. It must be explicitly enabled by the user. It is enabled by setting 
+`enabled: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of `metrics-server` instances are being deployed for for multi-nodes clusters and 1 for SNO clusters in the openshift-monitoring namespace.
+
+Once metrics-server is fully functional (after 5 scrape cycles done by metrics-server) we will remove `prometheus-adapter` resources fully.
+
+### Risks and Mitigations
+
+When considering equal scrape intervals, both metrics-server and prometheus-adapter have a comparable impact. This similarity arises from the fact that both applications perform a kubelet scrape once per instance within the designated scrape interval.
```

Comment:
We will actually be doing one additional scrape with metrics-server enabled.

With prometheus-adapter only Prometheus scrapes kubelet, but with metrics-server you need both Prometheus and metrics-server to scrape the kubelet in order to also get container metrics in Prometheus.